### PR TITLE
shell: Make the super-user-indicator apply to the current machine

### PR DESCRIPTION
**Pull request by mvollmer**
- [x] #13663

### CODE REVIEW

**Comment by mvollmer** [comment]
This needs updated tests.

**Comment by mvollmer** [comment]
Tests are up-to-date.

File: pkg/shell/superuser.jsx [hunk 1]
```diff
@@ -143,10 +143,23 @@ class LockDialog extends React.Component {
 }
 
 export class SuperuserDialogs extends React.Component {
-    constructor() {
+    constructor(props) {
         super();
```
Comment (martinpitt) [inline pkg/shell/superuser.jsx:147]: If you accept props, they should be propagated to the parent constructor.

File: test/verify/check-superuser [hunk 2]
```diff
@@ -359,40 +359,34 @@ class TestSuperuserDashboard(MachineCase):
         b.click('#troubleshoot-dialog button:contains("Connect")')
         b.wait_popdown('troubleshoot-dialog')
 
-        # The superuser indicator should be on the Overview page and not in the Shell
+        # The superuser indicator in the Shell should apply to machine2
 
-        b.wait_text("#super-user-indicator", "")
-        b.enter_page("/system", host="10.111.113.2")
-        b.wait_in_text(".ct-overview-header-actions", "Administrative access")
```
Comment (martinpitt) [inline test/verify/check-superuser:366]: Is showing the status in the Overview still a thing in some cases? I don't see that PR removing it. Then the test should assert that it's not present?

File: pkg/shell/superuser.jsx (hunk 1 above)
Comment (mvollmer) [reply pkg/shell/superuser.jsx:147]: Good point, we don't need props in the constructor.  But also see: https://stackoverflow.com/questions/30571875/whats-the-difference-between-super-and-superprops-in-react-when-using-e

File: test/verify/check-superuser (hunk 2 above)
Comment (mvollmer) [reply test/verify/check-superuser:366]: Yes, when a new machine is on the dashboard of an old machine.  And we indeed need for more tests for that.

File: pkg/shell/superuser.jsx (hunk 1 above)
Comment (martinpitt) [reply pkg/shell/superuser.jsx:147]: Interesting! I suppose I'm thinking too much in Python or C++ terms, where `this.props` must be set by the constructor. Nevermind then, I learned something new :)

Comment (mvollmer) [reply pkg/shell/superuser.jsx:147]: Yeah, I would understand it if one had to _always_ pass props to super(), but it seems to be optional, and then it must be always optional, no matter what args your own constructor takes.

**Pull request merged**